backend/compact-connect/lambdas/cloudfront-csp/index.js

```diff
@@ -1,5 +1,337 @@
+
+    return environmentUrls;
+};
+
+/**
+ * Helper to escape CSP src keywords.
+ * @param  {string} keyword The standard keyword value.
+ * @return {string}         The escaped keyword value.
+ */
+const srcKeywordEscape = (keyword) => {
+    let escaped = '';
+
+    if (keyword && typeof keyword === 'string') {
+        escaped = `'${keyword}'`.toLowerCase();
+    }
+
+    return escaped;
+};
+
+/**
+ * Helper to automatically escape and prep CSP src keyword values (by reference).
+ * @param  {Array<string>} srcList       The CSP src list.
+ * @param  {string}        [listName=''] Optional src list name for logging.
+ */
+const srcKeywordsEscape = (srcList, listName = '') => {
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#keyword_values
+    const srcKeywordsConfig = [
+        { value: 'self', isAllowed: true },
+        { value: 'none', isAllowed: true },
+        { value: 'strict-dynamic', isAllowed: true },
+        { value: 'report-sample', isAllowed: true },
+        { value: 'inline-speculation-rules', isAllowed: true },
+        { value: 'unsafe-inline', isAllowed: true },
+        { value: 'unsafe-eval', isAllowed: false },
+        { value: 'unsafe-hashes', isAllowed: false },
+        { value: 'wasm-unsafe-eval', isAllowed: false },
+    ];
+    const srcKeywords = srcKeywordsConfig.map((config) => config.value.toLowerCase());
+
+    if (Array.isArray(srcList)) {
+        srcList.forEach((srcItem, idx) => {
+            const isString = typeof srcItem === 'string';
+
+            if (!isString) {
+                srcList[idx] = '';
+            } else {
+                const srcItemLowerCase = srcItem.toLowerCase();
+
+                if (srcKeywords.includes(srcItemLowerCase)) {
+                    const keywordConfig = srcKeywordsConfig.find((config) => srcItemLowerCase === config.value.toLowerCase());
+
+                    if (keywordConfig) {
+                        if (!keywordConfig.isAllowed) {
+                            console.warn(`${listName} ${srcItem} keyword is not allowed in srcKeywordsConfig policy. We likely should not be using this keyword for security reasons.`.trim());
+                            srcList[idx] = '';
+                        } else {
+                            srcList[idx] = srcKeywordEscape(srcItem);
+                        }
+                    }
+                }
+            }
+        });
+    }
+};
+
+/**
+ * Helper to build a CSP src group list string from input params.
+ * @param  {string}        name src name for the CSP group.
+ * @param  {Array<string>} list The static src list for the CSP group.
+ * @return {string}             The prepped src list string;
+ */
+const buildSrcString = (name = '', list = []) => {
+    let src = [];
+    let srcString = '';
+
+    if (Array.isArray(list)) {
+        src = src.concat(list);
+    }
+
+    srcKeywordsEscape(src, name);
```

Comment:
Didn't catch this last time, but why not just assert (or similar) that list is an array of strings and remove all the checks in `buildSrcString` and `srcKeywordsEscape`? You may have a good reason but am curious because it seems it would simplify the code significantly by removing many edge cases.

Reply:
This `buildSrcString()` function had some extra cruft that I removed. Basically remnants of when multiple optional lists could be passed in. No longer necessary. The rest I feel kinda keeps each function durable & able to stand on its own.

Reply:
Makes sense, thanks for the explanation!